### PR TITLE
Smaller BookInfo images

### PR DESCRIPTION
**Pull request by ijsnellf**
**What this PR does / why we need it**: reduces the size of the BookInfo images by ~700mb for ratings, details and productpage. This makes the experience nicer by reducing the amount of time it takes to download all the images (especially for slow connections).

I also refactored some of the productpage code so that it uses standard Python idioms. I can move it to a different PR if needed.

**Which issue this PR fixes** fixes https://github.com/istio/istio/issues/1002

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by rshriram** [comment]
please update bookinfo/kube/bookinfo.yaml to point to your updated copy of bookinfo images (non sidecar ones). We can fix the bookinfo yaml again in a subsequent PR after pushing to istio docker hub

**Comment by ijsnellf** [comment]
Need to verify this works in k8s

**Comment by ijsnellf** [comment]
Verified it works with k8s

**Comment by ijsnellf** [comment]
@rshriram: I modified bookinfo.yaml to point to updated copies of the images.

File: samples/bookinfo/kube/bookinfo-add-serviceaccount.yaml [hunk 1]
```diff
@@ -18,7 +18,7 @@ spec:
       serviceAccountName: bookinfo-productpage
       containers:
       - name: productpage
-        image: istio/examples-bookinfo-productpage-v1:0.2.3
+        image: ijsnellf/examples-bookinfo-productpage-v1:0.2.7
```
Comment (ldemailly) [inline samples/bookinfo/kube/bookinfo-add-serviceaccount.yaml]: nope

**Comment by linsun** [comment]
thank you @ijsnellf for using the slim images and switch to use the .deb package. very nice!

**Comment by ldemailly** [comment]
why was this merged despite my request for change ?
cc @smawson

**Pull request merged**